helper.mac.fish

```diff
@@ -77,22 +81,98 @@ function findRequiredMinMacOS
 end
 
 function opensslVersion
-  set -l oversion $argv[1]
+  set -gx OPENSSL_VERSION $argv[1]
+end
 
-  if test "$oversion" = ""
-    set -e OPENSSL_VERSION
-    return 0
+function downloadOpenSSL
+  if test -z $OPENSSL_VERSION
+    echo "Need OPENSSL_VERSION global variable!"
+    return 1
   end
 
-  switch $oversion
-    case '1.0.2'
-      set -gx OPENSSL_VERSION $oversion
+  set -l directory $WORKDIR/work/openssl
+  set -l url https://www.openssl.org/source/openssl-$OPENSSL_VERSION.tar.gz  
+  mkdir -p $directory
+  cd $directory
+  echo "Downloading sources to $directory from URL: $url"
+  curl -LO $url
+  rm -rf openssl-$OPENSSL_VERSION
+  tar -xzvf openssl-$OPENSSL_VERSION.tar.gz
+end
 
-    case '1.1.1'
-      set -gx OPENSSL_VERSION $oversion
+function buildOpenSSL
+  if test "$OPENSSL_SOURCE_DIR" = ""
+    echo "OPENSSL_SOURCE_DIR is not defined. Please download source codes before building OpenSSL!"
```

Comment:
How do I download?
it seems "run fish> downloadOpenSSL" should be printed out here to improve the message

Reply:
Fixed.